app/hooks/use-message-events.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from 'react';
+import { Subject } from 'rxjs';
+
+export const messages$ = new Subject<any>();
+
+const handler = (e: MessageEvent) => {
+  if (e.origin !== 'file://' && e.source !== window) return;
+  messages$.next(e.data);
+};
+window.addEventListener('message', handler);
+
+export function useMessageEvents() {
+  const [message, setMessage] = useState<unknown>();
+  useEffect(() => {
+    const sub = messages$.subscribe(message => setMessage(message));
+    return sub.unsubscribe;
+  }, []);
```

Comment:
There is some naming conflicts here, two instances of `message`, also you're dependency of `setMessage` isn't defined in the useEffect hook
```suggestion
  }, [setMessage]);
```

Reply:
Are you sure that's necessary? Per https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects the example doesn't include the useState `set` method in the dep array. 

>  all values from the component scope (such as props and state) that change over time and that are used by the effect.
It doesn't change over time, right?

Reply:
You're right!
> React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.